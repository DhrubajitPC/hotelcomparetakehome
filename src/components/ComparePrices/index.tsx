import React, { FC } from "react";
import { useComparisons } from "../../hooks";

type Props = {
  hotelId: number;
};



const ComparePrices: FC<Props> = ({ hotelId }) => {
  const comparisons = useComparisons(hotelId);
  return comparisons.length > 0 ? (
    <div
      style={{
        display: "flex",
        flexDirection: "column",
      }}
    >
      <h4>The Competition</h4>
      {comparisons.map((c) => {
        return (
          <div
            style={{
              display: "flex",
              flexDirection: "row",
              justifyContent: "space-between",
            }}
          >
            <p>{c.name}</p>
            <p>
              {c.currency} {c.price}
            </p>
          </div>
        );
      })}
    </div>
  ) : null;
};

export default ComparePrices;
