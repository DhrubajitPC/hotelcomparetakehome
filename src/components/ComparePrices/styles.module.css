.container {
	display: flex;
	flex-direction: column;
}

.comparisonItem {
	display: flex;
	justify-content: space-between;
	box-shadow: 0 1px 2px -2px var(--black);
}

.comparisonItem:last-child{
	box-shadow: none;
}

.strikeThrough {
	position: relative;
}
.strikeThrough::before {
	position: absolute;
	content: '';
	border: 1px solid var(--red);
	transform: rotate(-5deg);
	top:50%;
	left:0;
	right:0;
}