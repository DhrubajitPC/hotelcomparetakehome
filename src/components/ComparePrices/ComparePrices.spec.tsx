import { render, screen } from "@testing-library/react";
import ComparePrices from ".";

const mockComparions = [
  {
    name: "expedia",
    price: 100,
    currency: "USD",
  },
  {
    name: "Us",
    price: 110,
    currency: "USD",
  },
  {
    name: "airbnb",
    price: 120,
    currency: "USD",
  },
];

jest.mock("../../hooks", () => ({
  __esModule: true,
  useComparisons: () => mockComparions,
}));

describe("#ComparePrices", () => {
  it("should render correctly", () => {
    const { container } = render(<ComparePrices hotelId={1} />);
    expect(container).toMatchSnapshot();
    expect(screen.getByText(/The Competition/i)).toBeInTheDocument();
    expect(screen.getByText(/120/)).toHaveClass("strikeThrough");
  });
});
