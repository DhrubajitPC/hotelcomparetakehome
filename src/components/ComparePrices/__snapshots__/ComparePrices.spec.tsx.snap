// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#ComparePrices should render correctly 1`] = `
<div>
  <div
    class="container"
  >
    <h4>
      The Competition
    </h4>
    <div
      class="comparisonItem"
    >
      <p>
        expedia
      </p>
      <p
        class=""
      >
        USD
         
        100
      </p>
    </div>
    <div
      class="comparisonItem"
    >
      <p>
        Us
      </p>
      <p
        class=""
      >
        USD
         
        110
      </p>
    </div>
    <div
      class="comparisonItem"
    >
      <p>
        airbnb
      </p>
      <p
        class="strikeThrough"
      >
        USD
         
        120
      </p>
    </div>
  </div>
</div>
`;
